File: backend/src/main/java/com/smartcart/service/UserService.java
package com.smartcart.service;

import com.smartcart.entity.User;
import com.smartcart.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserService {

    @Autowired
    private UserRepository userRepository;

    // Save User
    public User saveUser(User user) {
        return userRepository.save(user);
    }

    // Get All Users
    public List<User> getAllUsers() {
        return userRepository.findAll();
    }

    // Get User By Id
    public User getUserById(Long id) {
        return userRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("User not found"));
    }

    // Get User By Email
    public User getUserByEmail(String email) {
        return userRepository.findByEmail(email)
                .orElseThrow(() -> new RuntimeException("User not found"));
    }

    // Delete User
    public void deleteUser(Long id) {
        userRepository.deleteById(id);
    }

    // Check Email Exists
    public boolean emailExists(String email) {
        return userRepository.existsByEmail(email);
    }
}

File: Reactfrontend/src/services/userService.js
import api from "./api";

// Get All Users
export const getAllUsers = () => {
    return api.get("/users");
};

// Get User By Id
export const getUserById = (id) => {
    return api.get(`/users/${id}`);
};

// Get User By Email
export const getUserByEmail = (email) => {
    return api.get(`/users/email/${email}`);
};

// Create User
export const createUser = (user) => {
    return api.post("/users", user);
};

// Update User
export const updateUser = (id, user) => {
    return api.put(`/users/${id}`, user);
};

// Delete User
export const deleteUser = (id) => {
    return api.delete(`/users/${id}`);
};